src/main/java/uk/gov/hmcts/reform/pip/data/management/service/LocationService.java

```diff
@@ -168,12 +174,19 @@ public Collection<Location> uploadLocations(MultipartFile locationList) {
                         LocationType.valueOfCsv(locationCsv.getProvenanceLocationType()))));
 
                 try {
+                    validationService.containsForbiddenCharacter(location.getName());
+                    validationService.containsForbiddenCharacter(location.getWelshName());
                     savedLocations.add(locationRepository.save(location));
                 } catch (DataIntegrityViolationException e) {
                     log.error(writeLog(String.format(
                         "Record with ID %d not saved. The location name '%s' or Welsh location name '%s' already "
                             + "exists", location.getLocationId(), location.getName(), location.getWelshName()
                     )));
+                } catch (PayloadValidationException e) {
```

Comment:
I dont think u need to pass back the exception to the calling method just to add a log. You can log it in ValidationService before throwing the exception. The other exception in this method DataIntegrityViolationException is different as it is not thrown by us but by spring boot